WordPress/Classes/ViewRelated/Site Creation/Site Intent/SiteIntentViewController.swift

```diff
@@ -110,6 +110,7 @@ class SiteIntentViewController: CollapsableHeaderViewController {
     @objc
     private func closeButtonTapped(_ sender: Any) {
         SiteCreationAnalyticsHelper.trackSiteIntentCanceled()
+        RootViewCoordinator.shared.reloadPostSiteCreationCancellationIfNeeded()
```

Comment:
Right now we always reload the UI if some conditions are met, even if the UI shouldn't be reloaded.
I think we can just do this:
```suggestion
        RootViewCoordinator.shared.isSiteCreationActive = false
        RootViewCoordinator.shared.reloadUIIfNeeded(blog: blog)
```
We can either pass a nil value for the blog or fetch the current blog for consistency. There shouldn't be a need for the isWordPress or blog count condition.

Reply:
I tried this but `reloadUIIfNeeded` does not reload still. iirc `guard newUIType != oldUIType` was falling into else. That's why I force the reload. However we don't reload in any case. The new function checks if the user has blogs and is on WordPress app. Since this function is only invoked from `SiteIntentViewController`, and that can only happen if the user has specifically 0 blogs and taps "close" on the first site creation screen, it will run on this very specific case and nothing else. Not on Jetpack app, not if the user has already a site.

Reply:
I tried it locally and it works as expected:

https://github.com/wordpress-mobile/WordPress-iOS/assets/25306722/4eaae7e8-e950-439d-bb1f-3cb1f8fa81f1

Reply:
@alpavanoglu I tried it locally and it works as expected:

https://github.com/wordpress-mobile/WordPress-iOS/assets/25306722/4eaae7e8-e950-439d-bb1f-3cb1f8fa81f1

Reply:
I tested it and it does work indeed. Perhaps I mistakenly tried the same account and the overlay was already shown once. I pushed the changes